### ISSUE TITLE
Making trend uncertainty optional

### ISSUE THREAD

**Issue by asu04** [facebook/prophet#1007]
Hi!

I was wondering if it was possible or even recommended to have functionality that reduces the portion of the uncertainty due to trend change to 0. I think my use case is that I actually want the ability to detect anomalies in "real time". Prediction intervals seem like a good place to start: as new observations come in, I can compare the observations to the generated intervals. However, it seems like based on my reading of the paper that I would probably then not want to factor in potential trend changes in the future into my uncertainty; in fact, I'd want to just make the assumption that there will be no more trend changes in the future since the whole idea is that if the trend changes enough, the observations should start being out of bounds of the predicted intervals.

I guess that might be somewhat equivalent to training on the time range since the last change point and setting 0 change points, but I do want to take advantage of any seasonal patterns and fixed holiday effects that are modeled as a result of of having the full history, which I believe to be very stable in time, as well as modelling the trend changes in history.

I'm not totally sure this is even the right approach to what I'm trying to accomplish. Another thought that occurred to me is to maybe bootstrap the model's residuals on the training data - but was wondering if you had any thoughts.

Thanks!

**Comment by bletham** [comment]
This is a pretty specific use case so I'm pretty hesitant to add an interface for this, but I think you can effectively get this result with only a little extra work outside fbprophet.

The main estimate `y_hat` is the MAP estimate, and does not incorporate the trend uncertainty in any way. So to remove trend uncertainty, you just have to manually construct uncertainty intervals while leaving out trend uncertainty. If you don't do MCMC, this is very easy: The only uncertainty comes from the constant noise term. You can get the fitted noise standard deviation as:
```
noise_sd = m.params['sigma_obs'] * m.y_scale
```
So a 2-SD uncertainty interval would be
```
yhat(t) +/- 1.96 * noise_sd
```

If you do MCMC then you'll need to actually simulate from the posterior samples for the seasonal components. For this you would use this function, but replace `trend = self.sample_predictive_trend(df, iteration)` with just using the constant future trend:
https://github.com/facebook/prophet/blob/ebb304680d126e5d3a83d4ef9eb1f50cb5965273/python/fbprophet/forecaster.py#L1392

**Comment by asu04** [comment]
Thanks for the response! That makes sense - I'm also wondering if it seems reasonable to get a sense of the actual coverage probabilities of the PIs based on the CV results. This seems more intuitive and empirical to me than just extending the model forward, maybe except that the CV errors are probably not quite independent, though that probably isn't too bad if you choose the CV parameters carefully?

**Comment by germayneng** [comment]
sorry , i do have a question: For what you did above, is it considered a prediction interval or confidence interval? I believe it is a prediction interval right?

https://robjhyndman.com/hyndsight/intervals/

**Comment by bletham** [comment]
@asu04 I think that would be totally reasonable.

**Comment by bletham** [comment]
@germayneng It is a prediction interval, since it incorporates the noise term. If we assume known trend and seasonality, then all of the prediction uncertainty comes from the noise term, and so its intervals define the prediction interval. If we assume that the future trend is unknown but has a certain generative model (that described here: https://facebook.github.io/prophet/docs/uncertainty_intervals.html#uncertainty-in-the-trend) then we can simulate from that generative process and then add the noise to get prediction intervals.

**Comment by asu04** [comment]
Thanks for all your help @bletham 

@germayneng Correct me if I'm wrong, but I think the reason that it's more confusing in this case is because the equivalent of the traditional "uncertainty in parameter estimation" (e.g. the slope parameter), which is what gives confidence intervals, isn't separated here from the prior. If you just take the MAP estimation (and therefore the predicted yhat values), you assume 0 uncertainty not only in future trend changes, but also the parameter estimate of the slope itself.
